app/src/main/java/com/android/sample/ui/filter/FilterPage.kt

```diff
@@ -0,0 +1,317 @@
+ * @param unit The unit of the slider.
+ * @param range The range of the slider.
+ * @param updateRange The function to update the range.
+ */
+@Composable
+fun ValueRangeSlider(
+    modifier: Modifier,
+    name: String,
+    min: Float,
+    max: Float,
+    unit: String,
+    range: FloatRange,
+    updateRange: (Float, Float) -> Unit,
+) {
+  // Update the range to the [min, max] if it is unbounded
+  if (range.isLimited()) {
+    range.update(min, max)
+  }
+  // Mutable state to store the slider position
+
+  var sliderPosition by remember { mutableStateOf(range.min..range.max) }
+  Column(modifier = Modifier.padding(16.dp), horizontalAlignment = Alignment.CenterHorizontally) {
+    Text(
+        text = name,
+        style = MaterialTheme.typography.titleMedium,
+        modifier = Modifier.align(Alignment.Start))
+
+    Spacer(modifier = Modifier.height(8.dp))
+
+    // Display selected min and max values
+    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
+      Text(
+          "${sliderPosition.start.toInt()} " + unit,
+          style = MaterialTheme.typography.bodyMedium,
+          color = MaterialTheme.colorScheme.onPrimary)
+      Text(
+          "${sliderPosition.endInclusive.toInt()} " + unit,
+          style = MaterialTheme.typography.bodyMedium,
+          color = MaterialTheme.colorScheme.onPrimary)
+    }
+
+    Spacer(modifier = Modifier.height(8.dp))
+
+    // Range slider with minimum 0 and maximum 120 minutes
+    RangeSlider(
+        modifier = modifier.fillMaxWidth(),
+        value = sliderPosition,
+        onValueChange = {
+          sliderPosition = it
+          updateRange(it.start, it.endInclusive)
+        },
+        valueRange = min..max,
+        colors =
+            SliderDefaults.colors(
+                thumbColor = MaterialTheme.colorScheme.primary,
+                activeTrackColor = MaterialTheme.colorScheme.primary.copy(alpha = 0.4f),
+                inactiveTrackColor = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.2f)))
+  }
+}
+
+/**
+ * Composable function to display checkboxes for difficulty levels.
+ *
+ * @param recipesViewModel The view model to manage recipes.
+ */
+@SuppressLint("MutableCollectionMutableState", "StateFlowValueCalledInComposition")
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun CheckboxDifficulty(recipesViewModel: RecipesViewModel) {
+  // List of difficulty names
+
+  val difficultyLevels = remember {
+    mutableStateMapOf(
+        Difficulty.Easy to false, Difficulty.Medium to false, Difficulty.Hard to false)
+  }
+  val orderedDifficulties = listOf(Difficulty.Easy, Difficulty.Medium, Difficulty.Hard)
+
+  if (recipesViewModel.filter.value.difficulty != Difficulty.Undefined) {
+    difficultyLevels[recipesViewModel.filter.value.difficulty] = true
+  }
+
+  Column(modifier = Modifier.padding(16.dp), horizontalAlignment = Alignment.CenterHorizontally) {
+    Text(
+        text = "Difficulty",
```

Comment:
Great job handling the magic strings in most areas, it looks like just this one was missed. Easy to overlook! Updating it for consistency would keep everything clean and maintainable.